From 19294cb703b7c1e4bc87366810382416f90045a0 Mon Sep 17 00:00:00 2001
From: =?UTF-8?q?Alejandro=20Gonz=C3=A1lez?=
 <alejandro.gonzalez.correo@gmail.com>
Date: Sun, 20 Sep 2020 23:05:02 +0200
Subject: [PATCH 1/1] package/qprint: new package
MIME-Version: 1.0
Content-Type: text/plain; charset=UTF-8
Content-Transfer-Encoding: 8bit

qprint is a small utility that handles encoding and decoding arbitrary
binary data in the Quoted-Printable format, which was introduced along
MIME.

As stated in the package description, although MUAs nowadays usually
take care of doing this encoding themselves, there are some use cases
where doing it manually is needed, and providing a command-line program
to do is amenable to shell scripts.

Signed-off-by: Alejandro González <alejandro.gonzalez.correo@gmail.com>
---
 DEVELOPERS                 |  1 +
 package/Config.in          |  1 +
 package/qprint/Config.in   | 18 ++++++++++++++++++
 package/qprint/qprint.hash |  3 +++
 package/qprint/qprint.mk   | 13 +++++++++++++
 5 files changed, 36 insertions(+)
 create mode 100644 package/qprint/Config.in
 create mode 100644 package/qprint/qprint.hash
 create mode 100644 package/qprint/qprint.mk

diff --git a/DEVELOPERS b/DEVELOPERS
index 057592b8e5..e6f2695e86 100644
--- a/DEVELOPERS
+++ b/DEVELOPERS
@@ -119,6 +119,7 @@ F:	package/bird/
 F:	package/glorytun/
 
 N:	Alejandro González <alejandro.gonzalez.correo@gmail.com>
+F:	package/qprint/
 F:	package/watchdog/
 
 N:	Aleksander Morgado <aleksander@aleksander.es>
diff --git a/package/Config.in b/package/Config.in
index 0282a2069f..0bea80356c 100644
--- a/package/Config.in
+++ b/package/Config.in
@@ -2351,6 +2351,7 @@ comment "Utilities"
 	source "package/logsurfer/Config.in"
 	source "package/pdmenu/Config.in"
 	source "package/pinentry/Config.in"
+	source "package/qprint/Config.in"
 	source "package/ranger/Config.in"
 	source "package/rtty/Config.in"
 	source "package/screen/Config.in"
diff --git a/package/qprint/Config.in b/package/qprint/Config.in
new file mode 100644
index 0000000000..edfb92a5f3
--- /dev/null
+++ b/package/qprint/Config.in
@@ -0,0 +1,18 @@
+config BR2_PACKAGE_QPRINT
+	bool "qprint"
+	help
+	  qprint is a program that encodes and decodes binary data in
+	  the Quoted-Printable encoding, defined as a part of the MIME
+	  specification in RFC 1521. This encoding allows for efficient
+	  transfer of mostly ASCII data over transports that are only
+	  7-bit clean, like SMTP without 8BITMIME or BINARYMIME
+	  extensions. It also allows the transfer of long lines of text
+	  by wrapping them every 76 characters.
+
+	  Usually, a sophisticated MUA will take care of automatically
+	  choosing the best encodings for an e-mail. Nevertheless, this
+	  utility may be useful when a MTA (e.g. msmtp) is used without
+	  a MUA, or if using this encoding for other purposes is
+	  desired.
+
+	  https://www.fourmilab.ch/webtools/qprint/
diff --git a/package/qprint/qprint.hash b/package/qprint/qprint.hash
new file mode 100644
index 0000000000..0412cf84c5
--- /dev/null
+++ b/package/qprint/qprint.hash
@@ -0,0 +1,3 @@
+# Locally computed
+sha256  ffa9ca1d51c871fb3b56a4bf0165418348cf080f01ff7e59cd04511b9665019c  qprint-1.1.tar.gz
+sha256  3bd4089942977a357fed0d3c84b6ea877000c20c7c344c2feabf4b10db9f7495  COPYING
diff --git a/package/qprint/qprint.mk b/package/qprint/qprint.mk
new file mode 100644
index 0000000000..e34fba54b9
--- /dev/null
+++ b/package/qprint/qprint.mk
@@ -0,0 +1,13 @@
+################################################################################
+#
+# qprint
+#
+################################################################################
+
+QPRINT_VERSION = 1.1
+QPRINT_SITE = https://www.fourmilab.ch/webtools/qprint
+QPRINT_INSTALL_TARGET_OPTS = DESTDIR=$(TARGET_DIR) install-bin
+QPRINT_LICENSE = Public Domain
+QPRINT_LICENSE_FILES = COPYING
+
+$(eval $(autotools-package))
-- 
2.28.0

